SimpleUploadServer/index.js

```diff
@@ -0,0 +1,55 @@
+const { ApolloServer, gql, GraphQLUpload } = require("apollo-server");
+const promisesAll = require("promises-all");
+const { db, processUpload } = require("./file-utils");
+
+const typeDefs = gql`
+  type File {
+    id: ID!
+    path: String!
+    filename: String!
+    mimetype: String!
+  }
+  type Query {
+    uploads: [File]
+  }
+  type Mutation {
+    singleUpload(file: Upload!): File!
+    multipleUpload(files: [Upload!]!): [File!]!
```

Comment:
@hwillson OK so I think I can see how to add this to the typedefs but I want to do something like `multipleParameterUpload(singleFile: Upload!, multipleFiles: [Upload!]!): [File!]!` - I'm not sure how to add that to the resolvers though.

Reply:
@designatednerd You could do something like:

```js
async multipleParameterUpload(obj, { singleFile, multipleFiles }) {
  const { resolve, reject } = await promisesAll.all(
    [singleFile, ...multipleFiles].map(processUpload)
  );

  if (reject.length)
    reject.forEach(({ name, message }) =>
      // eslint-disable-next-line no-console
      console.error(`${name}: ${message}`)
    );

  return resolve;
}
```

Reply:
Mostly working, but I think there's something up with this: 

```js
[singleFile, ...multipleFiles].map(processUpload)
```

The files I'm getting back when it resolves are only the ones uploaded via `multipleFiles`

Reply:
@designatednerd That's odd! If you drop a `console.log(singleFile)` in here:

```js
async multipleParameterUpload(obj, { singleFile, multipleFiles }) {
  console.log(singleFile);
  ...
```

Is a `singleFile` value logged when the resolver is hit?

Reply:
Yes, though there's another error in there:

```
a.txt
TypeError: createReadStream is not a function
```

Reply:
@designatednerd it looks like the server side resolver is getting passed a string instead of an `Upload`. That's why the resolver is dumping out `a.txt` as the value of `singleFile`, instead of something like:

```js
Promise {
  {
    filename: 'a.txt',
    mimetype: 'application/octet-stream',
    encoding: '7bit',
    createReadStream: [Function: createReadStream]
  }
}
```

Since it's getting a string, when we try to access the `createReadStream` function on the upload, we can't find it. Is it possible that something in `UploadMultipleFilesToDifferentParametersMutation` is preventing the full `Upload` type from being sent over properly, and is instead sending a string?

Reply:
Definitely possible. Will look at this, thank you!